### PR TITLE
fix: issue where could not use interfaces

### PR DESCRIPTION
**Pull request by unparalleled-js**
### What I did

fixes: #45 

The issue is that vyper does not like the interfaces when they are passed in as source files. It seems to work fine when they are imported and not passed in as source files.

### How I did it

* Skip over sources that have a parent of `interfaces/` so no interfaces files are ever passed in as source files

### How to verify it

Can compile a project with interfaces in it.
Can use interfaces in contracts.

### Checklist

- [x] Passes all linting checks (pre-commit and CI jobs)
- [x] New test cases have been added and are passing
- [x] Documentation has been updated
- [x] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

**Comment by unparalleled-js** [comment]
Sounds like this is not 100% the solution we want, and we may need to make changes to ape core.
Discussions to be had!

File: README.md [hunk 1]
```diff
@@ -38,6 +38,16 @@ ape compile
 
 The `.vy` files in your project will compile into `ContractTypes` that you can deploy and interact with in Ape.
 
+### Interfaces
+
+You can not compile interface source files directly.
+Thus, you must place interface files in a directory named `interfaces` in your `contracts_folder` e.g. `contracts/interfaces/IFace.vy`.
```
Comment (helloibis) [inline README.md:44]: Basically any files that only define interfaces right (since those wouldn't generate any opcodes)?

**Review by helloibis — APPROVED** [review]

**Comment by unparalleled-js** [comment]
@charles-cooper and @fubuloubu I am wondering your thoughts on this since you know more about the vyper sides of things.

Is this solution okay now, at least for a short term? I don't think any harm could come of it unless you put non-interface material in a folder names interfaces. Does that ever happen?

**Comment by fubuloubu** [comment]
> Is this solution okay now, at least for a short term? I don't think any harm could come of it unless you put non-interface material in a folder names interfaces. Does that ever happen?

This is more of a "framework expectation" sort of thing. Like is anyone else using it in this way? Or using it in a way that directly conflicts with this usage. Brownie had a feature where a separate interfaces folder could be defined that lives outside of the contracts folder, but you can import things in. I think that's the feature people expect. Is there a way to provide this feature in a different way that is a lot easier and less weird/error-prone to use? Worth some research

**Comment by charles-cooper** [comment]
> This is more of a "framework expectation" sort of thing. Like is anyone else using it in this way? Or using it in a way that directly conflicts with this usage. Brownie had a feature where a separate interfaces folder could be defined that lives outside of the contracts folder, but you can import things in. I think that's the feature people expect. Is there a way to provide this feature in a different way that is a lot easier and less weird/error-prone to use? Worth some research

agreed, the vyper does not really know about project layout, it only knows about files.

if i'm understanding correctly, the solution is to segregate compilation - only compile contracts in the `contracts/` folder and never those in the `interfaces/` folder. this makes sense. vyper interface files are not currently compile-able as standalone contracts, they can only be imported, so i think your solution works.

(as an aside, i am redoing how interfaces work in vyper, so we may need to revisit this at some point in the future)

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**